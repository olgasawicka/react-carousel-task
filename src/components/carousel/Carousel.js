import React from "react";
import PropTypes from "prop-types";
import CarouselStyles from "./CarouselStyles";
import Slide from "./Slide";

const Carousel = ({ slidesData }) => {
  return (
    <CarouselStyles>
      {slidesData.map((slide) => (
        <Slide key={slide.id} slide={slide} />
      ))}
    </CarouselStyles>
  );
};

Carousel.propTypes = {
  slidesData: PropTypes.array.isRequired,
};

export default Carousel;
