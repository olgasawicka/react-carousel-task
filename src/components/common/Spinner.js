import React from "react";
import GridLoader from "react-spinners/GridLoader";

const Spinner = () => (
  <div data-testid="spinner">
    <GridLoader color={"var(--sky)"} />
  </div>
);

export default Spinner;
