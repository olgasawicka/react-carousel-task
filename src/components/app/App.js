import React, { createContext, useEffect, useState } from "react";
import data from "../../data/data";
import Carousel from "../carousel/Carousel";
import AppStyles, { GlobalStyles } from "./AppStyles";
import Spinner from "../common/Spinner";

export const AppContext = createContext({
  slideWidth: 0,
  setSlideWidth: () => {},
});

const App = () => {
  const [loading, setLoading] = useState(true);
  const [slidesData, setSlidesData] = useState(null);
  const [slideWidth, setSlideWidth] = useState(0);

  useEffect(() => {
    setSlidesData(data);
  }, []);

  useEffect(() => {
    slidesData && setLoading(false);
  }, [slidesData]);

  return (
    <>
      <GlobalStyles />
      <AppContext.Provider
        value={{
          slideWidth,
          setSlideWidth,
        }}
      >
        <AppStyles>
          {loading && !slidesData ? (
            <Spinner />
          ) : (
            <Carousel slidesData={slidesData} />
          )}
        </AppStyles>
      </AppContext.Provider>
    </>
  );
};

export default App;
