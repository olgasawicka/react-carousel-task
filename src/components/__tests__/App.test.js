import { cleanup, render } from "@testing-library/react";
import React from "react";
import mockData from "../../data/data";
import App from "../app/App";
import Carousel from "../carousel/Carousel";
import Spinner from "../common/Spinner";

afterEach(cleanup);

describe("App Component", () => {

  const pauseStub = jest
    .spyOn(window.HTMLMediaElement.prototype, "pause")
    .mockImplementation(() => {});

  it("should take a snapshot", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it("renders Spinner component", () => {
    const { getByTestId } = render(<Spinner />);
    expect(getByTestId("spinner")).toBeInTheDocument();
  });

  it("renders Carousel component", () => {
    const { getByTestId } = render(<Carousel slidesData={mockData} />);
    expect(getByTestId("carousel")).toBeInTheDocument();

    expect(pauseStub).toHaveBeenCalled();
    pauseStub.mockRestore();
  });
});
