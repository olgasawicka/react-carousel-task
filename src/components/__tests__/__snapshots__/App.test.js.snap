// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should take a snapshot 1`] = `
<div>
  <div
    class="sc-dlnjwi csZBcG"
  >
    <div
      class="sc-bdnxRM bYicFh"
      data-testid="carousel"
    >
      <div
        class="slider"
      >
        <div
          class="slider-wrapper"
          data-testid="slider-wrapper"
          style="transform: translateX(calc(-16.666666666666668% - 0px));"
        >
          <div
            class="sc-gtsrHT iJNmgx prev"
            data-testid="slide"
          >
            <img
              alt="Slide border"
              class="slide-border"
              src="mockup_blank_dark.png"
            />
            <div
              class="slide-bg"
            >
              <img
                alt="Slide background"
                src="slide_bg.jpg"
              />
            </div>
            <div
              class="thumb-wrapper"
            >
              <img
                alt="Book Cover"
                src="./assets/images/alice_in_wonderland.jpg"
              />
            </div>
            <div
              class="btn-wrapper"
            >
              <button
                data-testid="play-btn"
                disabled=""
              >
                <svg
                  height="72px"
                  version="1.1"
                  viewBox="0 0 72 72"
                  width="72px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <defs>
                    <filter
                      filterUnits="objectBoundingBox"
                      height="140.0%"
                      id="filter-1"
                      width="140.0%"
                      x="-20.0%"
                      y="-20.0%"
                    >
                      <feoffset
                        dx="0"
                        dy="2"
                        in="SourceAlpha"
                        result="shadowOffsetOuter1"
                      />
                      <fegaussianblur
                        in="shadowOffsetOuter1"
                        result="shadowBlurOuter1"
                        stdDeviation="2"
                      />
                      <fecolormatrix
                        in="shadowBlurOuter1"
                        result="shadowMatrixOuter1"
                        type="matrix"
                        values="0 0 0 0 0.0862745098   0 0 0 0 0.031372549   0 0 0 0 0.160784314  0 0 0 0.2 0"
                      />
                      <femerge>
                        <femergenode
                          in="shadowMatrixOuter1"
                        />
                        <femergenode
                          in="SourceGraphic"
                        />
                      </femerge>
                    </filter>
                  </defs>
                  <g
                    fill="none"
                    fill-rule="evenodd"
                    id="img-/-i_play"
                    stroke="none"
                    stroke-width="1"
                  >
                    <g
                      filter="url(#filter-1)"
                      id="icons-/-player-/-i_play_circle"
                      transform="translate(6.000000, 6.000000)"
                    >
                      <path
                        d="M29.999938,6.28830321e-13 C46.5599787,6.28830321e-13 60,13.4399038 60,29.999908 C60,46.5600348 46.5599787,60 29.999938,60 C13.4400833,60 -1.70530257e-13,46.5600348 -1.70530257e-13,29.999908 C-1.70530257e-13,13.4399038 13.4400833,6.28830321e-13 29.999938,6.28830321e-13 Z M22.5,19.5054199 L22.5,40.6099366 C22.5,42.2707017 23.6311367,42.8793496 25.0264646,41.9714392 L40.9186037,31.6307595 C42.3189705,30.7195704 42.3139316,29.2584865 40.9186037,28.360014 L25.0264646,18.1268283 C23.6260978,17.2251112 22.5,17.8455723 22.5,19.5054199 Z"
                        fill="#FFFFFF"
                        id="Shape"
                      />
                      <path
                        d="M25.0264646,41.9714392 L40.9186037,31.6307595 C42.3189705,30.7195704 42.3139316,29.2584865 40.9186037,28.360014 L25.0264646,18.1268283 C23.6260978,17.2251112 22.5,17.8455723 22.5,19.5054199 L22.5,40.6099366 C22.5,42.2707017 23.6311367,42.8793496 25.0264646,41.9714392 Z"
                        fill="var(--sky)"
                        id="Path"
                        opacity="0.7"
                      />
                    </g>
                  </g>
                </svg>
              </button>
            </div>
          </div>
          <div
            class="sc-gtsrHT iJNmgx active"
            data-testid="slide"
          >
            <img
              alt="Slide border"
              class="slide-border"
              src="mockup_blank_dark.png"
            />
            <div
              class="slide-bg"
            >
              <img
                alt="Slide background"
                src="slide_bg.jpg"
              />
            </div>
            <div
              class="thumb-wrapper"
            >
              <img
                alt="Book Cover"
                src="./assets/images/japanese_fairy_tales.jpg"
              />
            </div>
            <div
              class="btn-wrapper"
            >
              <button
                data-testid="play-btn"
              >
                <svg
                  height="72px"
                  version="1.1"
                  viewBox="0 0 72 72"
                  width="72px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <defs>
                    <filter
                      filterUnits="objectBoundingBox"
                      height="140.0%"
                      id="filter-1"
                      width="140.0%"
                      x="-20.0%"
                      y="-20.0%"
                    >
                      <feoffset
                        dx="0"
                        dy="2"
                        in="SourceAlpha"
                        result="shadowOffsetOuter1"
                      />
                      <fegaussianblur
                        in="shadowOffsetOuter1"
                        result="shadowBlurOuter1"
                        stdDeviation="2"
                      />
                      <fecolormatrix
                        in="shadowBlurOuter1"
                        result="shadowMatrixOuter1"
                        type="matrix"
                        values="0 0 0 0 0.0862745098   0 0 0 0 0.031372549   0 0 0 0 0.160784314  0 0 0 0.2 0"
                      />
                      <femerge>
                        <femergenode
                          in="shadowMatrixOuter1"
                        />
                        <femergenode
                          in="SourceGraphic"
                        />
                      </femerge>
                    </filter>
                  </defs>
                  <g
                    fill="none"
                    fill-rule="evenodd"
                    id="img-/-i_play"
                    stroke="none"
                    stroke-width="1"
                  >
                    <g
                      filter="url(#filter-1)"
                      id="icons-/-player-/-i_play_circle"
                      transform="translate(6.000000, 6.000000)"
                    >
                      <path
                        d="M29.999938,6.28830321e-13 C46.5599787,6.28830321e-13 60,13.4399038 60,29.999908 C60,46.5600348 46.5599787,60 29.999938,60 C13.4400833,60 -1.70530257e-13,46.5600348 -1.70530257e-13,29.999908 C-1.70530257e-13,13.4399038 13.4400833,6.28830321e-13 29.999938,6.28830321e-13 Z M22.5,19.5054199 L22.5,40.6099366 C22.5,42.2707017 23.6311367,42.8793496 25.0264646,41.9714392 L40.9186037,31.6307595 C42.3189705,30.7195704 42.3139316,29.2584865 40.9186037,28.360014 L25.0264646,18.1268283 C23.6260978,17.2251112 22.5,17.8455723 22.5,19.5054199 Z"
                        fill="#FFFFFF"
                        id="Shape"
                      />
                      <path
                        d="M25.0264646,41.9714392 L40.9186037,31.6307595 C42.3189705,30.7195704 42.3139316,29.2584865 40.9186037,28.360014 L25.0264646,18.1268283 C23.6260978,17.2251112 22.5,17.8455723 22.5,19.5054199 L22.5,40.6099366 C22.5,42.2707017 23.6311367,42.8793496 25.0264646,41.9714392 Z"
                        fill="var(--sky)"
                        id="Path"
                        opacity="0.7"
                      />
                    </g>
                  </g>
                </svg>
              </button>
            </div>
          </div>
          <div
            class="sc-gtsrHT iJNmgx next"
            data-testid="slide"
          >
            <img
              alt="Slide border"
              class="slide-border"
              src="mockup_blank_dark.png"
            />
            <div
              class="slide-bg"
            >
              <img
                alt="Slide background"
                src="slide_bg.jpg"
              />
            </div>
            <div
              class="thumb-wrapper"
            >
              <img
                alt="Book Cover"
                src="./assets/images/the_secret_garden.jpg"
              />
            </div>
            <div
              class="btn-wrapper"
            >
              <button
                data-testid="play-btn"
                disabled=""
              >
                <svg
                  height="72px"
                  version="1.1"
                  viewBox="0 0 72 72"
                  width="72px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <defs>
                    <filter
                      filterUnits="objectBoundingBox"
                      height="140.0%"
                      id="filter-1"
                      width="140.0%"
                      x="-20.0%"
                      y="-20.0%"
                    >
                      <feoffset
                        dx="0"
                        dy="2"
                        in="SourceAlpha"
                        result="shadowOffsetOuter1"
                      />
                      <fegaussianblur
                        in="shadowOffsetOuter1"
                        result="shadowBlurOuter1"
                        stdDeviation="2"
                      />
                      <fecolormatrix
                        in="shadowBlurOuter1"
                        result="shadowMatrixOuter1"
                        type="matrix"
                        values="0 0 0 0 0.0862745098   0 0 0 0 0.031372549   0 0 0 0 0.160784314  0 0 0 0.2 0"
                      />
                      <femerge>
                        <femergenode
                          in="shadowMatrixOuter1"
                        />
                        <femergenode
                          in="SourceGraphic"
                        />
                      </femerge>
                    </filter>
                  </defs>
                  <g
                    fill="none"
                    fill-rule="evenodd"
                    id="img-/-i_play"
                    stroke="none"
                    stroke-width="1"
                  >
                    <g
                      filter="url(#filter-1)"
                      id="icons-/-player-/-i_play_circle"
                      transform="translate(6.000000, 6.000000)"
                    >
                      <path
                        d="M29.999938,6.28830321e-13 C46.5599787,6.28830321e-13 60,13.4399038 60,29.999908 C60,46.5600348 46.5599787,60 29.999938,60 C13.4400833,60 -1.70530257e-13,46.5600348 -1.70530257e-13,29.999908 C-1.70530257e-13,13.4399038 13.4400833,6.28830321e-13 29.999938,6.28830321e-13 Z M22.5,19.5054199 L22.5,40.6099366 C22.5,42.2707017 23.6311367,42.8793496 25.0264646,41.9714392 L40.9186037,31.6307595 C42.3189705,30.7195704 42.3139316,29.2584865 40.9186037,28.360014 L25.0264646,18.1268283 C23.6260978,17.2251112 22.5,17.8455723 22.5,19.5054199 Z"
                        fill="#FFFFFF"
                        id="Shape"
                      />
                      <path
                        d="M25.0264646,41.9714392 L40.9186037,31.6307595 C42.3189705,30.7195704 42.3139316,29.2584865 40.9186037,28.360014 L25.0264646,18.1268283 C23.6260978,17.2251112 22.5,17.8455723 22.5,19.5054199 L22.5,40.6099366 C22.5,42.2707017 23.6311367,42.8793496 25.0264646,41.9714392 Z"
                        fill="var(--sky)"
                        id="Path"
                        opacity="0.7"
                      />
                    </g>
                  </g>
                </svg>
              </button>
            </div>
          </div>
          <div
            class="sc-gtsrHT iJNmgx"
            data-testid="slide"
          >
            <img
              alt="Slide border"
              class="slide-border"
              src="mockup_blank_dark.png"
            />
            <div
              class="slide-bg"
            >
              <img
                alt="Slide background"
                src="slide_bg.jpg"
              />
            </div>
            <div
              class="thumb-wrapper"
            >
              <img
                alt="Book Cover"
                src="./assets/images/moby_dick.jpg"
              />
            </div>
            <div
              class="btn-wrapper"
            >
              <button
                data-testid="play-btn"
                disabled=""
              >
                <svg
                  height="72px"
                  version="1.1"
                  viewBox="0 0 72 72"
                  width="72px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <defs>
                    <filter
                      filterUnits="objectBoundingBox"
                      height="140.0%"
                      id="filter-1"
                      width="140.0%"
                      x="-20.0%"
                      y="-20.0%"
                    >
                      <feoffset
                        dx="0"
                        dy="2"
                        in="SourceAlpha"
                        result="shadowOffsetOuter1"
                      />
                      <fegaussianblur
                        in="shadowOffsetOuter1"
                        result="shadowBlurOuter1"
                        stdDeviation="2"
                      />
                      <fecolormatrix
                        in="shadowBlurOuter1"
                        result="shadowMatrixOuter1"
                        type="matrix"
                        values="0 0 0 0 0.0862745098   0 0 0 0 0.031372549   0 0 0 0 0.160784314  0 0 0 0.2 0"
                      />
                      <femerge>
                        <femergenode
                          in="shadowMatrixOuter1"
                        />
                        <femergenode
                          in="SourceGraphic"
                        />
                      </femerge>
                    </filter>
                  </defs>
                  <g
                    fill="none"
                    fill-rule="evenodd"
                    id="img-/-i_play"
                    stroke="none"
                    stroke-width="1"
                  >
                    <g
                      filter="url(#filter-1)"
                      id="icons-/-player-/-i_play_circle"
                      transform="translate(6.000000, 6.000000)"
                    >
                      <path
                        d="M29.999938,6.28830321e-13 C46.5599787,6.28830321e-13 60,13.4399038 60,29.999908 C60,46.5600348 46.5599787,60 29.999938,60 C13.4400833,60 -1.70530257e-13,46.5600348 -1.70530257e-13,29.999908 C-1.70530257e-13,13.4399038 13.4400833,6.28830321e-13 29.999938,6.28830321e-13 Z M22.5,19.5054199 L22.5,40.6099366 C22.5,42.2707017 23.6311367,42.8793496 25.0264646,41.9714392 L40.9186037,31.6307595 C42.3189705,30.7195704 42.3139316,29.2584865 40.9186037,28.360014 L25.0264646,18.1268283 C23.6260978,17.2251112 22.5,17.8455723 22.5,19.5054199 Z"
                        fill="#FFFFFF"
                        id="Shape"
                      />
                      <path
                        d="M25.0264646,41.9714392 L40.9186037,31.6307595 C42.3189705,30.7195704 42.3139316,29.2584865 40.9186037,28.360014 L25.0264646,18.1268283 C23.6260978,17.2251112 22.5,17.8455723 22.5,19.5054199 L22.5,40.6099366 C22.5,42.2707017 23.6311367,42.8793496 25.0264646,41.9714392 Z"
                        fill="var(--sky)"
                        id="Path"
                        opacity="0.7"
                      />
                    </g>
                  </g>
                </svg>
              </button>
            </div>
          </div>
          <div
            class="sc-gtsrHT iJNmgx"
            data-testid="slide"
          >
            <img
              alt="Slide border"
              class="slide-border"
              src="mockup_blank_dark.png"
            />
            <div
              class="slide-bg"
            >
              <img
                alt="Slide background"
                src="slide_bg.jpg"
              />
            </div>
            <div
              class="thumb-wrapper"
            >
              <img
                alt="Book Cover"
                src="./assets/images/around_the_world_in_80_days.jpg"
              />
            </div>
            <div
              class="btn-wrapper"
            >
              <button
                data-testid="play-btn"
                disabled=""
              >
                <svg
                  height="72px"
                  version="1.1"
                  viewBox="0 0 72 72"
                  width="72px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <defs>
                    <filter
                      filterUnits="objectBoundingBox"
                      height="140.0%"
                      id="filter-1"
                      width="140.0%"
                      x="-20.0%"
                      y="-20.0%"
                    >
                      <feoffset
                        dx="0"
                        dy="2"
                        in="SourceAlpha"
                        result="shadowOffsetOuter1"
                      />
                      <fegaussianblur
                        in="shadowOffsetOuter1"
                        result="shadowBlurOuter1"
                        stdDeviation="2"
                      />
                      <fecolormatrix
                        in="shadowBlurOuter1"
                        result="shadowMatrixOuter1"
                        type="matrix"
                        values="0 0 0 0 0.0862745098   0 0 0 0 0.031372549   0 0 0 0 0.160784314  0 0 0 0.2 0"
                      />
                      <femerge>
                        <femergenode
                          in="shadowMatrixOuter1"
                        />
                        <femergenode
                          in="SourceGraphic"
                        />
                      </femerge>
                    </filter>
                  </defs>
                  <g
                    fill="none"
                    fill-rule="evenodd"
                    id="img-/-i_play"
                    stroke="none"
                    stroke-width="1"
                  >
                    <g
                      filter="url(#filter-1)"
                      id="icons-/-player-/-i_play_circle"
                      transform="translate(6.000000, 6.000000)"
                    >
                      <path
                        d="M29.999938,6.28830321e-13 C46.5599787,6.28830321e-13 60,13.4399038 60,29.999908 C60,46.5600348 46.5599787,60 29.999938,60 C13.4400833,60 -1.70530257e-13,46.5600348 -1.70530257e-13,29.999908 C-1.70530257e-13,13.4399038 13.4400833,6.28830321e-13 29.999938,6.28830321e-13 Z M22.5,19.5054199 L22.5,40.6099366 C22.5,42.2707017 23.6311367,42.8793496 25.0264646,41.9714392 L40.9186037,31.6307595 C42.3189705,30.7195704 42.3139316,29.2584865 40.9186037,28.360014 L25.0264646,18.1268283 C23.6260978,17.2251112 22.5,17.8455723 22.5,19.5054199 Z"
                        fill="#FFFFFF"
                        id="Shape"
                      />
                      <path
                        d="M25.0264646,41.9714392 L40.9186037,31.6307595 C42.3189705,30.7195704 42.3139316,29.2584865 40.9186037,28.360014 L25.0264646,18.1268283 C23.6260978,17.2251112 22.5,17.8455723 22.5,19.5054199 L22.5,40.6099366 C22.5,42.2707017 23.6311367,42.8793496 25.0264646,41.9714392 Z"
                        fill="var(--sky)"
                        id="Path"
                        opacity="0.7"
                      />
                    </g>
                  </g>
                </svg>
              </button>
            </div>
          </div>
          <div
            class="sc-gtsrHT iJNmgx"
            data-testid="slide"
          >
            <img
              alt="Slide border"
              class="slide-border"
              src="mockup_blank_dark.png"
            />
            <div
              class="slide-bg"
            >
              <img
                alt="Slide background"
                src="slide_bg.jpg"
              />
            </div>
            <div
              class="thumb-wrapper"
            >
              <img
                alt="Book Cover"
                src="./assets/images/uncle_toms_cabin.jpg"
              />
            </div>
            <div
              class="btn-wrapper"
            >
              <button
                data-testid="play-btn"
                disabled=""
              >
                <svg
                  height="72px"
                  version="1.1"
                  viewBox="0 0 72 72"
                  width="72px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <defs>
                    <filter
                      filterUnits="objectBoundingBox"
                      height="140.0%"
                      id="filter-1"
                      width="140.0%"
                      x="-20.0%"
                      y="-20.0%"
                    >
                      <feoffset
                        dx="0"
                        dy="2"
                        in="SourceAlpha"
                        result="shadowOffsetOuter1"
                      />
                      <fegaussianblur
                        in="shadowOffsetOuter1"
                        result="shadowBlurOuter1"
                        stdDeviation="2"
                      />
                      <fecolormatrix
                        in="shadowBlurOuter1"
                        result="shadowMatrixOuter1"
                        type="matrix"
                        values="0 0 0 0 0.0862745098   0 0 0 0 0.031372549   0 0 0 0 0.160784314  0 0 0 0.2 0"
                      />
                      <femerge>
                        <femergenode
                          in="shadowMatrixOuter1"
                        />
                        <femergenode
                          in="SourceGraphic"
                        />
                      </femerge>
                    </filter>
                  </defs>
                  <g
                    fill="none"
                    fill-rule="evenodd"
                    id="img-/-i_play"
                    stroke="none"
                    stroke-width="1"
                  >
                    <g
                      filter="url(#filter-1)"
                      id="icons-/-player-/-i_play_circle"
                      transform="translate(6.000000, 6.000000)"
                    >
                      <path
                        d="M29.999938,6.28830321e-13 C46.5599787,6.28830321e-13 60,13.4399038 60,29.999908 C60,46.5600348 46.5599787,60 29.999938,60 C13.4400833,60 -1.70530257e-13,46.5600348 -1.70530257e-13,29.999908 C-1.70530257e-13,13.4399038 13.4400833,6.28830321e-13 29.999938,6.28830321e-13 Z M22.5,19.5054199 L22.5,40.6099366 C22.5,42.2707017 23.6311367,42.8793496 25.0264646,41.9714392 L40.9186037,31.6307595 C42.3189705,30.7195704 42.3139316,29.2584865 40.9186037,28.360014 L25.0264646,18.1268283 C23.6260978,17.2251112 22.5,17.8455723 22.5,19.5054199 Z"
                        fill="#FFFFFF"
                        id="Shape"
                      />
                      <path
                        d="M25.0264646,41.9714392 L40.9186037,31.6307595 C42.3189705,30.7195704 42.3139316,29.2584865 40.9186037,28.360014 L25.0264646,18.1268283 C23.6260978,17.2251112 22.5,17.8455723 22.5,19.5054199 L22.5,40.6099366 C22.5,42.2707017 23.6311367,42.8793496 25.0264646,41.9714392 Z"
                        fill="var(--sky)"
                        id="Path"
                        opacity="0.7"
                      />
                    </g>
                  </g>
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
